src/interface/blas3/backend/default_cpu.hpp

```diff
@@ -120,6 +120,48 @@ _gemm(sb_handle_t& sb_handle, index_t _M, index_t _N, index_t _K,
   }
 }
 
+// Half Configurations
+template <bool _t_a, bool _t_b, bool s_a, bool s_b, bool is_beta_zero,
+          typename sb_handle_t, typename container_0_t, typename container_1_t,
+          typename container_2_t, typename element_t, typename index_t>
+typename std::enable_if<is_half<element_t>::value,
+                        typename sb_handle_t::event_t>::type
+_gemm(sb_handle_t& sb_handle, index_t _M, index_t _N, index_t _K,
+      element_t _alpha, container_0_t _a, index_t _lda, index_t _stridea,
+      container_1_t _b, index_t _ldb, index_t _strideb, element_t _beta,
+      container_2_t _c, index_t _ldc, index_t _stridec, index_t batch_size,
+      gemm_batch_type_t batch_type,
+      const typename sb_handle_t::event_t& _dependencies) {
+  // Unused configuration cases
+  if constexpr (s_a || s_b) {
```

Comment:
Thanks for your PR, @OuadiElfarouki.

I think you are excluding the symmetric matrices because they are not considered in case of the half datatype.

If this is the case, could you please add a comment specifying what are the supported routines?

Reply:
Indeed  @pgorlani. We have the option to disable building those kernels for CPU which we know doesn't support `fp16`, but when using oneMKL we can't directly do that given the header only integration within oneMKL and the fact that `aspect::fp16` is only known at runtime. This means that the added Gemm configurations for the default CPU target are merely a temporary **placeholder** (until such support is brought to CPUs eventually), and they were consequently never tested as they're skipped by the unit-tests. This change comes in-line with the patch on oneMKL (not published yet) : https://github.com/OuadiElfarouki/oneMKL/pull/1

Reply:
Hi @OuadiElfarouki, it seems that the portBLAS backend for oneMKL checks for `aspect::f16`, see https://github.com/oneapi-src/oneMKL/blob/3339418fa1318377c65815e6f85aced9efa46e9e/src/blas/backends/portblas/portblas_common.hpp#L196.